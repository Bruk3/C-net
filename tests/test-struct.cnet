struct person{
  string name;
 int age;
};

struct person p1 = new struct person; 
p1.name = "Hello /n world";
p1.age = 12; 

struct couple{
  struct person p1;
};

struct couple q1 = new struct person; 

q1.p1.age =14;