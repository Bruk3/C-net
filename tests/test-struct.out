STRUCT 
ID(person)
LBRACE 
STRING 
ID(name)
SEMI
INT 
ID(age)
SEMI
RBRACE
SEMI 
STRUCT 
ID(person) 
ID(p1)
ASSIGN
NEW 
STRUCT 
ID(person)
SEMI 
ID(p1)
DOT
ID(name)
ASSIGN 
STRLIT(Hello /n world)
SEMI 
ID(p1)
DOT
ID(age)
ASSIGN 
INTLIT(12)
SEMI 
STRUCT 
ID(couple)
LBRACE 
STRUCT 
ID(person)
ID(p1)
SEMI
RBRACE
SEMI 
STRUCT 
ID(couple)
ID(q1)
ASSIGN
NEW
STRUCT
ID(person)
SEMI 
ID(q1)
DOT
ID(p1)
DOT
ID(age)
ASSIGN 
INTLIT(14)
SEMI
EOF